### PR TITLE
feat(marketpace): add new tool

### PR DESCRIPTION
**Pull request by slaveeks**
1. Added button on marketplace to add tool
2. Added Add Tool page
3. Added service, repository and application service methods to add tool

image

### CODE REVIEW

File: src/application/i18n/messages/en.json [hunk 1]
```diff
@@ -48,6 +48,26 @@
     "title": "Marketplace",
     "listOfTools": "Tools",
     "uninstallTool": "Uninstall",
-    "installTool": "Install"
+    "installTool": "Install",
+    "addTool": "Add Tool"
+  },
+  "newTool": {
```
Comment (neSpecc) [inline src/application/i18n/messages/en.json:54]: move it under the marketplace section

File: src/application/router/routes.ts [hunk 2]
```diff
@@ -6,6 +6,7 @@ import NoteSettings from '@/presentation/pages/NoteSettings.vue';
 import ErrorPage from '@/presentation/pages/Error.vue';
 import Marketplace from '@/presentation/pages/Marketplace.vue';
 import type { RouteRecordRaw } from 'vue-router';
+import AddTool from '@/presentation/pages/AddTool.vue';
```
Comment (neSpecc) [inline src/application/router/routes.ts:9]: lets move all marketplace pages under "marketplace" folder
```suggestion
import AddTool from '@/presentation/pages/marketplace/AddTool.vue';
```

File: src/application/router/routes.ts [hunk 3]
```diff
@@ -68,6 +69,11 @@ const routes: RouteRecordRaw[] = [
     path: `/marketplace`,
     component: Marketplace,
   },
+  {
+    name: 'newTool',
+    path: `/marketplace/newTool`,
```
Comment (neSpecc) [inline src/application/router/routes.ts:74]:
```suggestion
    path: `/marketplace/add`,
```

File: src/domain/entities/EditorTool.ts [hunk 4]
```diff
@@ -48,3 +53,8 @@ export interface EditorToolWithUserBinding extends EditorTool {
    */
   isInstalled: boolean;
 }
+
+/**
+ * Add new tool to the marketplace
+ */
+export type AddNewTool = Omit<EditorTool, 'userId' | 'id'>;
```
Comment (neSpecc) [inline src/domain/entities/EditorTool.ts:60]: strange name

File: src/application/i18n/messages/en.json (hunk 1 above)
Comment (slaveeks) [reply src/application/i18n/messages/en.json]: done

File: src/domain/entities/EditorTool.ts (hunk 4 above)
Comment (slaveeks) [reply src/domain/entities/EditorTool.ts]: renamed

File: src/domain/entities/EditorTool.ts [hunk 5]
```diff
@@ -48,3 +53,8 @@ export interface EditorToolWithUserBinding extends EditorTool {
    */
   isInstalled: boolean;
 }
+
+/**
+ * Add new tool to the marketplace
```
Comment (neSpecc) [inline src/domain/entities/EditorTool.ts:58]:
```suggestion
 * Tool creation attributes
```

**Review by neSpecc — APPROVED** [review]

**Review by e11sy — APPROVED** [review]

**Pull request merged**